### PR TITLE
DS-504 Implement ButtonGroup

### PR DESCRIPTION
**Pull request by david-cho-lerat-sonarsource**

### CODE REVIEW

File: src/components/buttons/ButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import styled from '@emotion/styled';
+import { ReactNode } from 'react';
+
+export enum ButtonGroupAlign {
+  Left = 'left',
+  Right = 'right',
+}
+
+export interface ButtonGroupProps {
```
Comment (gregaubert) [inline src/components/buttons/ButtonGroup.tsx:29]: Let's add a className

File: src/components/buttons/ButtonGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  align?: ButtonGroupAlign;
+  children: ReactNode;
+  isCombined?: boolean;
+}
+
+export function ButtonGroup({
```
Comment (gregaubert) [inline src/components/buttons/ButtonGroup.tsx:35]: Let's forward ref and forward the remaining props too

File: src/components/buttons/ButtonGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  flex: 1 0 auto;
+  gap: var(--echoes-dimension-space-100);
+  justify-content: ${({ align }) => (align === ButtonGroupAlign.Left ? 'flex-start' : 'flex-end')};
+  width: 100%;
+`;
+
+const StyledCombinedButtonGroup = styled(StyledButtonGroup)`
+  gap: 0;
+
+  & button {
```
Comment (gregaubert) [inline src/components/buttons/ButtonGroup.tsx:59]: If you rebase on the last version of my PR you can do `& ${ButtonStyled} {...` instead.

File: src/components/buttons/ButtonGroup.tsx (hunk 2 above)
Comment (david-cho-lerat-sonarsource) [reply src/components/buttons/ButtonGroup.tsx:35]: where should the ref go? on the container itself?

Comment (gregaubert) [reply src/components/buttons/ButtonGroup.tsx:35]: yeah, we don't have a better alternative anyway

Comment (david-cho-lerat-sonarsource) [reply src/components/buttons/ButtonGroup.tsx:35]: what remaining props could there be in this case?

Comment (david-cho-lerat-sonarsource) [reply src/components/buttons/ButtonGroup.tsx]: I'm wondering what the use case would be for the forwarded ref?

Comment (gregaubert) [reply src/components/buttons/ButtonGroup.tsx]: Making sure our ButtonGroup is compatible with our other Echoes components that rely on props and ref forwarding.

Comment (gregaubert) [reply src/components/buttons/ButtonGroup.tsx]: I don't see a use case right now. I just think it's good practice to make sure all our exported components can have a ref attached to them.
We might have usages where we need to know the size or positioning of the group.

Comment (david-cho-lerat-sonarsource) [reply src/components/buttons/ButtonGroup.tsx:31]: as discussed on Slack, as there's no use case for now and it's just a "structural" component with no inherent behavior, and it's not used through Redux/Mantine, let's keep it simple.

**Pull request merged**